### ISSUE TITLE
0.99.7 cannot built on centos.

### ISSUE THREAD

**Issue by weiby3** [async-rs/async-std#249]
hello world project from readme, use `cargo +beta run`, and got:
```
   Compiling async-std v0.99.7
error[E0599]: no method named `initializer` found for type `std::pin::Pin<&mut R>` in the current scope
  --> /home/weiby3/.cargo/registry/src/code.aliyun.com-738b7dba08a2a41e/async-std-0.99.7/src/io/read/read_to_end.rs:67:20
   |
67 |                 rd.initializer().initialize(&mut g.buf[g.len..]);
   |                    ^^^^^^^^^^^ method not found in `std::pin::Pin<&mut R>`
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following trait defines an item `initializer`, perhaps you need to implement it:
           candidate #1: `std::io::Read`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0599`.
error: could not compile `async-std`.

To learn more, run the command again with --verbose.
```
rustc version: `rustc 1.39.0-beta.1 (968967007 2019-09-24)`
os: `centos 7`

**Comment by yoshuawuyts** [comment]
We've discussed this in chat, and we'll be putting out a `0.99.8` release ahead of schedule to address this. Thanks for reporting!

**Comment by edwardw** [comment]
Run into the same issue myself. Before the release of `0.99.8`, a temporary workaround is to pin futures crate versions oneself if possible by adding:

```toml
futures-core-preview = "=0.3.0-alpha.18"
futures-io-preview = "=0.3.0-alpha.18"
```

to project's dependencies.

**Comment by skade** [comment]
I published 0.99.8 with the fix mentioned above. Thanks for the report!
